ebcli/operations/commonops.py

```diff
@@ -964,7 +964,12 @@ def create_default_instance_profile(profile_name=iam_attributes.DEFAULT_ROLE_NAM
     """
     Create default elasticbeanstalk IAM profile and return its name.
     """
-    create_instance_profile(profile_name, iam_attributes.DEFAULT_ROLE_POLICIES)
+    region = aws.get_region_name()
+    if isinstance(region, str) and 'us-gov' in region:
+        policies = iam_attributes.DEFAULT_ROLE_POLICIES_US_GOV
+    else:
```

Comment:
Thanks for contributing this change, @mccahan. Is this the only issue w.r.t. partition name you're seeing?

I think this might need to be a more holistic change than just the if/else in this portion of code. We use the standard ARN format in a couple of other places [in the CLI](https://github.com/search?q=repo%3Aaws%2Faws-elastic-beanstalk-cli%20%22arn%3Aaws%22&type=code). I'll work with someone on the team to investigate.

Reply:
Hey @rahulrajaram this is the only one I hit personally, but there are probably more - I'm running on an account that already had existing EB applicaitons so it didn't have to create the S3 bucket or instance roles so I didn't test. This probably applies to China as well, e.g. https://github.com/aws/aws-elastic-beanstalk-cli/blob/8c9ef5ccdff3805e6f6c5d5a8164d6e9048d5d58/ebcli/operations/commonops.py#L1148-L1152 which I modeled the check after.

I did look a little at your search results and I can confirm that at least some of the Option namespaces are consistent across the partitions:
![Screenshot 2025-06-29 at 19 57 44@2x](https://github.com/user-attachments/assets/a9b3f09b-0b07-462e-a6f7-c838afbae5c0)

Reply:
Okay.

I think for the time being, it makes sense to unblock you. One request -- could you actually undo the changes to CHANGES.rst. We'll take care of it.

Reply:
Thanks, I will merge it on Monday